cpp/src/fil/fil.cu

```diff
@@ -915,12 +1012,31 @@ void from_treelite(const raft::handle_t& handle, forest_t* pforest,
 void from_treelite(const raft::handle_t& handle, forest_t* pforest,
                    ModelHandle model, const treelite_params_t* tl_params) {
   const tl::Model& model_ref = *(tl::Model*)model;
-  model_ref.Dispatch([&handle, pforest, tl_params](const auto& model_inner) {
+  model_ref.Dispatch([&](const auto& model_inner) {
     // model_inner is of the concrete type tl::ModelImpl<T, L>
     from_treelite(handle, pforest, model_inner, tl_params);
   });
 }
 
+// allocates caller-owned char* using malloc()
+template <typename T, typename L, typename N>
```

Comment:
Also, with 3 types, single-letter type names become much harder to interpret. Consider having more descriptive type names, e.g. `threshold_t`, `leaf_t` or `node_t`.